Restore the {ocamlc,ocamlopt}_{cflags,cppflags} config vars

shindere opened the pull request:
These variables have been renamed by #13201 but should actually have
been preserved so that the use of `ocamlc -config` remains
backward-compatible.

anmonteiro reviewed (approved):
Thank you

MisterDA commented on Changes:
```diff
@@ -312,7 +312,7 @@ _______________
   Miod Vallat and Samuel Hym)
 
-* #13201: Fix and speedup builds with TSan.
+- #13201j #13244: Fix and speedup builds with TSan.
```
```suggestion
- #13201 #13244: Fix and speedup builds with TSan.
```

MisterDA reviewed (approved):
typo

shindere committed:
Restore the {ocamlc,ocamlopt}_{cflags,cppflags} config vars

These variables have been renamed by #13201 but should actually have been
preserved so that the use of `ocamlc -config` remains backward-compatible.

shindere commented:
thanks a lot for the prompt review and approval.

Antonin Décimo (2024/06/18 09:38 -0700):
> ```suggestion
> - #13201 #13244: Fix and speedup builds with TSan.
> ```
 
Thanks. Actually, the `j` was supposed to be a ','. It has now been fixed.

OlivierNicole reviewed (approved):
Approving on behalf of @anmonteiro, @MisterDA and @dustanddreams. I note that we may one day want to design a way to deprecate the old variables, but that is a longer discussion and for a later PR.

shindere commented:
Thanks a lot for the approval.

Such a deprecation mechanism would indeed be helpful.

Perhaps emiting a warning on stderr could work. To be tested on opam, I
guess.

emillon commented:
> Perhaps emiting a warning on stderr could work.

I don't think that's possible in the current interface, since users do `ocamlc -config | grep $NAME: | cut -d: -f2` or similar, so `ocamlc` doesn't know what options are used.

shindere commented:
OK thanks! So I'm guessing that the only option we are left with will be
to `opam grep` for the old names and fix what it reports.

Octachron commented:
Concerning the deprecation policy and backward-compatibility guarantees, I think that the `Config` is one of the compiler output that could benefit from using the structured diagnostic versioning policy described in https://github.com/ocaml/RFCs/pull/45 .
